### ISSUE TITLE
unexpect `class` rendered with merge

### ISSUE THREAD

**Issue by csbun** [developit/preact#260]
I used [preact-render-to-string](https://github.com/developit/preact-render-to-string) as a server-side rendering tool and rerender on the client-side with merge.

```jsx
import { h, render, Component } from 'preact';
/** @jsx h */

class Cmp extends Component {
  render() {
    return <p id="cmp" className="my-cmp">Cmp</p>;
  }
}

class App extends Component {
  render() {
    return (<div id="my-app">
      <p id="before">Before</p>
      <Cmp />
      <p id="after-1">After1</p>
      <p id="after-2">After2</p>
      <p id="after-3">After3</p>
    </div>);
  }
}
```

- server-side rendering
```jsx
import { h } from 'preact';
import render from 'preact-render-to-string';
const ssrHtml = render(<App />);
res.send(`
  <div>Preact</div>
  <div id="container">${ssrHtml}</div>
  <div id="log"></div>
`);
```

- client-side rendering
```jsx
import { h, render } from 'preact';
render(
  <App />,
  document.getElementById('container'),
  document.getElementById('my-app')
);
```

Here is the [full example](http://codepen.io/csbun/pen/RRqKGZ) on codepen.

As the result comes out, there is an unexpect `class` on `p#after-1`:
Server-side Rendering:
```html
<div id="my-app"><p id="before">Before</p><p id="cmp">Cmp</p><p id="after-1">After1</p><p id="after-2">After2</p><p id="after-3">After3</p></div>
```

Meanwhile Client-side Rendering:
```html
<div id="my-app"><p id="before">Before</p><p id="cmp">Cmp</p><p id="after-1" class="">After1</p><p id="after-2">After2</p><p id="after-3">After3</p></div>
```

And when I debug on Chrome DevTool, it seems `p#my-cmp`, `p#after-1`, `p#after-2`, `p#after-3` are all replaced. `removeOrphanedChildren` was call twice.

**Comment by developit** [comment]
Thanks for the great detailed report @csbun :)
I'm wondering if the is the same root cause as https://github.com/developit/preact/issues/257

**Comment by developit** [comment]
@csbun hmm - I'm seeing the same output in both cases. Was this with a specific browser?  I checked in Chrome.

**Comment by csbun** [comment]
Sorry I made an mistake on CodePen by missing `className="my-cmp"` in `class Cmp`. [Now](http://codepen.io/csbun/pen/RRqKGZ) it works out:

- Server-side Rendering

```html
<div id="my-app"><p id="before">Before</p><p id="cmp" class="my-cmp">Cmp</p><p id="after-1">After1</p><p id="after-2">After2</p><p id="after-3">After3</p></div>
```

- Client-side Rendering (merge)

```html
<div id="my-app"><p id="before">Before</p><p id="cmp" class="my-cmp">Cmp</p><p id="after-1" class="">After1</p><p id="after-2">After2</p><p id="after-3">After3</p></div>
```

**Comment by csbun** [comment]
I found it interesting that if I add an attribute `className="anything"` in `p#after-1`, the `class=""` will move to the next element `p#after-2`.

Example:

```jsx
class App extends Component {
  render() {
    return (<div id="my-app">
      <p id="before">Before</p>
      <Cmp />
      <p id="after-1" className="after-1">After1</p>
      <p id="after-2">After2</p>
      <p id="after-3">After3</p>
    </div>);
  }
}
```

Client-side Rendering (beautified just for good looking)

```html
<div id="my-app">
  <p id="before">Before</p>
  <p id="cmp" class="my-cmp">Cmp</p>
  <p id="after-1" class="after-1">After1</p>
  <p id="after-2" class="">After2</p>
  <p id="after-3">After3</p>
</div>
```

**Comment by developit** [comment]
Ahh - I think I see a little more where you were going with this.  Instead of recycling the `<p>` from SSR, `<Cmp>` should try to use it.  Makes perfect sense, I think there's a one-liner that might fix this!

**Comment by csbun** [comment]
Can't wait to see that.

**Comment by developit** [comment]
@csbun I believe this should now be fixed on master.

**Comment by developit** [comment]
@csbun Confirmed fixed in [6.0.0 (beta)](https://github.com/developit/preact/releases/tag/6.0.0):
http://codepen.io/developit/pen/BzgVvy?editors=0010#0

**Comment by csbun** [comment]
What a great Job!
Thanks! 谢谢!
